flutter-for-android.md

```diff
@@ -0,0 +1,1845 @@
+{% endprettify %}
+
+
+**How do I build custom Widgets?**
+
+Building custom widgets in Flutter is as simple as extending a Widget and 
+adding your custom logic. This is like Android where you would extend a View 
+and override the methods to customize it.
+
+Let's take a look at how to build a CustomButton that takes in the label in the 
+constructor and displays it
+
+<!-- skip -->
+{% prettify dart %}
+class CustomButton extends StatelessWidget {
+  final String label;
+  CustomButton(this.label);
+
+  @override
+  Widget build(BuildContext context) {
+    return new RaisedButton(onPressed: (){
+
+    }, child: new Text(label));
+  }
+}
+{% endprettify %}
+
+Then you can use this CustomButton just like you would with any other Widget
+
+<!-- skip -->
+{% prettify dart %}
+@override
+Widget build(BuildContext context) {
+  return  new Center(
+        child: new CustomButton("Hello"),
+  );
+}
+{% endprettify %}
+
+# Intents
+
+**What is the equivalent of an Intent in Flutter?**
+
+Flutter does not have the concept of Intents. In Android, there are two 
+main use-cases for Intents, One is to switch between Activities and another is 
+to invoke external(or internal) components of your app (Camera, Services etc).
+
+To switch between screens in Flutter you can access the router to draw 
+a new Widget.  There are two core concepts and classes for managing many 
+screens: Route and Navigator. A Route is an abstraction for a “screen” or 
+“page” of an app (think Activity), and a Navigator is a widget that manages 
+routes. A Navigator can push and pop routes to help a user move from screen to 
+screen.
+
+Like Android where you can declare your Activities inside the 
+AndroidManifest.xml, in Flutter you can pass in a Map of named routes to the 
+top level MaterialApp instance
+
+<!-- skip -->
+{% prettify dart %}
+ void main() {
+  runApp(new MaterialApp(
+    home: new MyAppHome(), // becomes the route named '/'
+    routes: <String, WidgetBuilder> {
+      '/a': (BuildContext context) => new MyPage(title: 'page A'),
+      '/b': (BuildContext context) => new MyPage(title: 'page B'),
+      '/c': (BuildContext context) => new MyPage(title: 'page C'),
+    },
+  ));
+}
+{% endprettify %}
+Then you can change to this route by getting an hold of the Navigator and 
+calling this named route.
+
+<!-- skip -->
+{% prettify dart %}
+Navigator.of(context).pushNamed('/b');
+{% endprettify %}
+
+The other popular use-case for Intents is to call external components such as a 
+Camera or File picker. For this, you would need to create a native platform 
+integration (or use an existing library)
+
+See [Flutter Plugins] to learn how to build a native platform integration.
+
+
+
+**How do I handle incoming Intents from external applications in Flutter?** 
+
+This can be achieved by registering the intent you want to handle in the 
+AndroidMainfest.xml to launch MainActivity (the default activity created by 
+Flutter)
+
+[code]
```

Comment:
do you still intend to add this?

Reply:
Oh oops, forgot this section, will add it during code review fixes right now